src/core_init_atmosphere/read_geogrid.c

```diff
@@ -20,55 +20,65 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifdef UNDERSCORE
-#define read_geogrid read_geogrid_
-#endif
-#ifdef DOUBLEUNDERSCORE
-#define read_geogrid read_geogrid__
-#endif
 
 #define GEOG_BIG_ENDIAN    0
 #define GEOG_LITTLE_ENDIAN 1
 
+/*  In Fortran, use the following as an interface for read_geogrid:
+
+ interface
+    subroutine read_geogrid(fname, rarray, nx, ny, nz, &
+                           & isigned, endian, scalefactor, wordsize, status) bind(C)
+       use iso_c_binding, only : c_char, c_int, c_float, c_ptr
+       character (c_char), dimension(*), intent(in) :: fname
+       type (c_ptr), value :: rarray
+       integer (c_int), intent(in), value :: nx
+       integer (c_int), intent(in), value :: ny
+       integer (c_int), intent(in), value :: nz
+       integer (c_int), intent(in), value :: isigned
+       integer (c_int), intent(in), value :: endian
+       real (c_float), intent(in), value :: scalefactor
+       integer (c_int), intent(in), value :: wordsize
+       integer (c_int), intent(inout) :: status
+    end subroutine read_geogrid
+ end interface
+
+ use iso_c_binding, only : c_char, c_int, c_float, c_ptr, c_loc
```

Comment:
Since `use` statements must come before `interface` statements in Fortran, I'd propose we either remove this line or move it above the example interface block.